### PR TITLE
feat(guides): transition to the new policies

### PR DESCRIPTION
**Pull request by lobkovilya**
Fix https://github.com/kumahq/kuma-website/issues/1729

### CODE REVIEW

File: app/_src/guides/transition-to-the-new-policies.md [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+spec:
+  sources:
+    - match:
+        kuma.io/service: "*"
+  destinations:
+    - match:
+        kuma.io/service: "*"
+  conf:
+    destination:
+      kuma.io/service: "*"' | kc apply -f-
```
Comment (Automaat) [inline app/_src/guides/transition-to-the-new-policies.md:127]:
```suggestion
      kuma.io/service: "*"' | kubectl apply -f-
```

File: app/_src/guides/transition-to-the-new-policies.md [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+2. Review the list of changes that're going to be created by the replacement policy.
+3. Remove `kuma.io/effect: shadow` label so that policy is applied in a normal mode.
+4. Observe metrics, traces and logs. If something goes wrong change policy's mode back to shadow and return to the step 2.
+If everything is fine then remove the old policies.
+
+### TrafficPermission -> MeshTrafficPermission
+
+1. Create a replacement policy for `app-to-redis` TrafficPermission and apply it with `kuma.io/effect: shadow` label:
+
+    ```sh
```
Comment (Automaat) [inline app/_src/guides/transition-to-the-new-policies.md:216]: this example have white space issues and does not work after coppying it

File: app/_src/guides/transition-to-the-new-policies.md [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+            tags: 
+              kuma.io/service: demo-app_kuma-demo_svc_5000
+          default:
+            action: Allow' | kubectl apply -f -
+    ```
+
+2. Check the list of changes for `redis_kuma-demo_svc_6379` pod in Envoy configuration using `kumactl`, `jq` and `jd`:
+
+    ```sh
+    kumactl inspect dataplane redis-8fcbfc795-twlst.kuma-demo --type=config --shadow --include=diff | jq '.diff' | jd -t patch2jd
```
Comment (Automaat) [inline app/_src/guides/transition-to-the-new-policies.md:241]: can you separate output to different block, right now when you use copy button you will also copy output

File: app/_src/guides/transition-to-the-new-policies.md [hunk 4]
```diff
@@ -0,0 +1,411 @@
...
+    Although the old TrafficPermission and the new MeshTrafficPermission currently coexist, the new policy completely overrides the old one.
+
+4. Observe the demo app behaves as expected. If everything goes well, we can safely remove TrafficPermission and conclude the transition.
+
+### Timeout -> MeshTimeout
+
+1. Create a replacement policy for `timeout-global` Timeout and apply it with `kuma.io/effect: shadow` label:
+
+    ```sh
+    echo 'apiVersion: kuma.io/v1alpha1
```
Comment (Automaat) [inline app/_src/guides/transition-to-the-new-policies.md:283]: this also have whitespace issues

File: app/_src/guides/transition-to-the-new-policies.md [hunk 5]
```diff
@@ -0,0 +1,411 @@
...
+          idleTimeout: 2h
+          http:
+            requestTimeout: 0s
+            streamIdleTimeout: 2h' | kubectl apply -f-
+    ```
+
+2. Check the list of changes for `redis_kuma-demo_svc_6379` pod in Envoy configuration using `kumactl`, `jq` and `jd`:
+
+    ```yaml
+    kumactl inspect dataplane redis-8fcbfc795-twlst.kuma-demo --type=config --shadow --include=diff | jq '.diff' | jd -t patch2jd
```
Comment (Automaat) [inline app/_src/guides/transition-to-the-new-policies.md:318]: again we can separate output

File: app/_src/guides/transition-to-the-new-policies.md [hunk 6]
```diff
@@ -0,0 +1,411 @@
...
+3. Remove the `kuma.io/effect: shadow` label.
+   Although the old Timeout and the new MeshTimeout currently coexist, the new policy completely overrides the old one.
+
+4. Observe the demo app behaves as expected. If everything goes well, we can safely remove Timeout and conclude the transition.
+
+### CircuitBreaker -> MeshCircuitBreaker
+
+1. Create a replacement policy for `cb-global` CircutBreaker and apply it with `kuma.io/effect: shadow` label:
+
+    ```sh
```
Comment (Automaat) [inline app/_src/guides/transition-to-the-new-policies.md:350]: another whitespace issue

File: app/_src/guides/transition-to-the-new-policies.md [hunk 7]
```diff
@@ -0,0 +1,411 @@
...
+3. Remove the `kuma.io/effect: shadow` label.
+   Although the old CircuitBreaker and the new MeshCircuitBreaker currently coexist, the new policy completely overrides the old one.
+
+4. Observe the demo app behaves as expected. If everything goes well, we can safely remove CircuitBreaker and conclude the transition.
+
+### TrafficRoute -> MeshTCPRoute
+
+It's safe to simply remove `route-all-default` TrafficRoute. 
+Traffic is going to flow through the system even if there are no either TrafficRoutes or MeshTCPRoutes/MeshHTTPRoutes.
+Ensure that TrafficRoute is the last policy deleted when removing older policies.
```
Comment (Automaat) [inline app/_src/guides/transition-to-the-new-policies.md:410]: can we make this bold? or as some warn, to make sure people see it? This seems important

File: app/_src/guides/transition-to-the-new-policies.md
Comment (Automaat) [inline app/_src/guides/transition-to-the-new-policies.md:1]: can we add `Next steps` section at the end of the guide like in other guides? To point people to new policies docs or any other useful resources?

File: app/_src/guides/transition-to-the-new-policies.md (hunk 2 above)
Comment (lobkovilya) [reply app/_src/guides/transition-to-the-new-policies.md:216]: right, not sure why it rendered like that, but I think I found a workaround

File: app/_src/guides/transition-to-the-new-policies.md (hunk 7 above)
Comment (lobkovilya) [reply app/_src/guides/transition-to-the-new-policies.md]: added warning block

File: app/_src/guides/transition-to-the-new-policies.md
Comment (lobkovilya) [reply app/_src/guides/transition-to-the-new-policies.md:1]: updated

**Review by Automaat — APPROVED** [review]

**Review by slonka — COMMENTED** [review]
In general great start, pointed out some things that might make sense to reconsider.

File: app/_src/guides/transition-to-the-new-policies.md [hunk 8]
```diff
@@ -0,0 +1,411 @@
...
+      failure:
+        requestVolume: 34
+        minimumHosts: 35
+        threshold: 36' | kubectl apply -f-
+```
+
+## Transition to the new policies
+
+Each type of policy is migrated separately; for example, once we have completely finished with the Timeouts, 
+we will proceed to the next policy type, CircuitBreakers.
```
[the remark this answers is not in the document]
Comment (lobkovilya) [reply app/_src/guides/transition-to-the-new-policies.md:204]: > Well theoretically it doesn't have to be migrated separately right? We just make it so in this guide right? 

I meant this as an advantage of our approach, so you don't have to commit to use all new policies at once. Maybe should rephrase it like:
```suggestion
Each type of policy can be migrated separately; for example, once we have completely finished with the Timeouts, 
we will proceed to the next policy type, CircuitBreakers.
```

> Also would it make sense to state it explicitly that the order does not matter?

Yes, I'll mention here that TrafficRoute should be the last.

[the remark this answers is not in the document]
Comment (lobkovilya) [reply app/_src/guides/transition-to-the-new-policies.md:204]: Also added:

> It's possible to migrate all policies at once, but small portions are preferable as they're easily reversible.

File: app/_src/guides/transition-to-the-new-policies.md [hunk 9]
```diff
@@ -0,0 +1,420 @@
...
+
+2.  Port-forward the service to the namespace on port 5000:
+
+    ```sh
+    kubectl port-forward svc/demo-app -n kuma-demo 5000:5000
+    ```
+
+3.  In a browser, go to [127.0.0.1:5000](http://127.0.0.1:5000) and increment the counter.
+
+### Enable Mutual TLS and Traffic Permissions
```
[the remark this answers is not in the document]
Comment (lobkovilya) [reply app/_src/guides/transition-to-the-new-policies.md:82]: I added mentioning of that at the beginning of `## Setup demo with old policies` section

File: app/_src/guides/transition-to-the-new-policies.md [hunk 10]
```diff
@@ -0,0 +1,420 @@
...
+        kuma.io/service: "*"
+  destinations:
+    - match:
+        kuma.io/service: "*"
+  conf:
+    connectTimeout: 21s
+    tcp: 
+      idleTimeout: 22s
+    http:
+      idleTimeout: 22s
```
[the remark this answers is not in the document]
Comment (lobkovilya) [reply app/_src/guides/transition-to-the-new-policies.md:153]: Unfortunately, the new MeshTimeout policy has a single place for the `tcp.idleTimeout` and `http.idleTimeout`. I guess the assumption was that the idleTimeouts are likely to be similar regardless of the protocol. TBH I find the diff less confusing when showcasing with equal idleTimeouts.

File: app/_src/guides/transition-to-the-new-policies.md [hunk 11]
```diff
@@ -0,0 +1,420 @@
...
+```
+
+### Deploy Timeouts
+
+```sh
+echo 'apiVersion: kuma.io/v1alpha1
+kind: Timeout
+mesh: default
+metadata:
+  name: timeout-global
```
[the remark this answers is not in the document]
Comment (lobkovilya) [reply app/_src/guides/transition-to-the-new-policies.md:140]: TBH I like the fact you can easily trace back the timeout in the policy by its value. I can add a note saying that by no means we're considering these good timeout values, it's done purely for demonstration.

File: app/_src/guides/transition-to-the-new-policies.md [hunk 12]
```diff
@@ -0,0 +1,411 @@
+---
+title: Transition to the new policies
+---
+
+{{site.mesh_product_name}} provides two set of policies to configure proxies. 
+There are old [source/destination](/docs/{{ page.version }}/policies/general-notes-about-kuma-policies/) policies that didn't live up to expectations and
+there are new [targetRef](/docs/{{ page.version }}/policies/targetref) policies that are designed to address the shortcomings of the old ones, 
+providing improved functionality and UX.
```
[the remark this answers is not in the document]
Comment (lobkovilya) [reply app/_src/guides/transition-to-the-new-policies.md:8]: I rephrased the intro, gave some credits to the s&d policies to sound less harsh :)

File: app/_src/guides/transition-to-the-new-policies.md [hunk 13]
```diff
@@ -0,0 +1,411 @@
...
+
+In this guide, we're going to setup a demo with old policies and then perform a transition to the new policies.
```
[the remark this answers is not in the document]
Comment (lobkovilya) [reply app/_src/guides/transition-to-the-new-policies.md:10]: Agreed, let's use "migration" everywhere

File: app/_src/guides/transition-to-the-new-policies.md [hunk 14]
```diff
@@ -0,0 +1,420 @@
...
+```
+
+## Transition to the new policies
+
+Each type of policy is migrated separately; for example, once we have completely finished with the Timeouts, 
+we will proceed to the next policy type, CircuitBreakers.
+
+The generalized migration process roughly consists of 4 steps:
+
+1. Create and deploy the replacement policies in [shadow mode](/docs/{{ page.version }}/policies/applying-policies/#applying-policies-in-shadow-mode).
```
[the remark this answers is not in the document]
Comment (lobkovilya) [reply app/_src/guides/transition-to-the-new-policies.md:208]: rephrased the first step

**Pull request merged**